packages/yew-macro/src/props/component.rs

```diff
@@ -156,17 +107,31 @@ impl ComponentProps {
                         .build()
                 }
             }
-            Self::With(with_props) => {
+            // Builder pattern is unnecessary in this case, since the base expression guarantees
+            // all values are initialized
+            Some(expr) => {
                 let ident = Ident::new("__yew_props", props_ty.span());
+                let set_props = self.props.iter().map(|Prop { label, value, .. }| {
+                    if is_string_literal(value) {
+                        // String literals should be implicitly converted into `String`
+                        quote_spanned! {value.span()=>
+                            #ident.#label = #value.into();
```

Comment:
In order for this to be hygenic, it shouldn't assume that the `Into` trait is in scope (even if it's part of the prelude).
```suggestion
                            #ident.#label = ::std::convert::Into::into(#value);
```

There are tests for this, but sadly none of the prop tests have been updated for this yet.